Fix QBCore Compatibility

RobertGrubb opened the pull request:
## Version 1.0.6

To understand the first bullet point, here is an example of an inventory add item call from latest QB Core (This was after the rename from qb-inventory to ps-inventory):

_Notice the last parameter passed_

```
exports['ps-inventory']:AddItem(src, Config.LuckyItem, random, false, false, 'qb-recyclejob:server:getItem')
```

- Updated "AddItem" method to take parameter for "reason" before "created" as this was causing the issue with the inventory to spit out the error `SCRIPT ERROR: @ps-inventory/server/main.lua.2544: attempt to sub a 'number' with a 'string'`. The "reason" was being passed as the "created" timestamp which caused "created" to come through as an unexpected string.
- During the "AddItem" method, "info" was being passed as an empty string at times, there is now a check to make sure the type is a table. If it is not, it will define it as an empty table so the assignment of "info.quality" will pass.
- Added `CreateShop` and `OpenShop` exports to `server.lua` so shops can correctly be created and opened.
- `RegisteredShops` table added at the top of the server file.

Previews (Latest QBCore) - Inventory opens, shops are functioning, images are working fine, and decay is good.
![image](https://github.com/user-attachments/assets/22dfaafd-fa81-4cde-b4c1-71466cfae7ea)
![image](https://github.com/user-attachments/assets/f0604946-6894-42a2-9738-b96c6028cdd0)

**NOTE: I'd appreciate it if we can get some eyes on this in terms of testing before merging just to confirm that all is well.**

Quenixx commented:
Hi Robert, Im kinda new to github, is there anyway to download these updates and update my own ps-inventory untill the pull is merged?

lenzh commented:
still recieved a error 
![image](https://github.com/user-attachments/assets/bdf97de6-7ee4-4bab-bd34-554a6238cac1)
this happened when opened a stash

lenzh commented:
ive changed the function ConvertQuality(item) to 
`function ConvertQuality(item)
    local StartDate = item.created
    local DecayRate = QBCore.Shared.Items[item.name:lower()]['decay'] ~= nil and QBCore.Shared.Items[item.name:lower()]['decay'] or 0.0
    if not DecayRate then
        DecayRate = 0.0
    end
    if not (type(StartDate) == "number") then
        if not (tonumber(StartDate)) then return 0 end
        StartDate = tonumber(StartDate)
    end
    local TimeExtra = math.ceil((TimeAllowed * DecayRate))
    local percentDone = 100 - math.ceil((((os.time() - StartDate) / TimeExtra) * 100))
    if DecayRate == 0 then
        percentDone = 100
    end
    if percentDone <= 0 then
        percentDone = 0
    end
    return percentDone
end`
might wanna use that because this fixed that issue for me

RobertGrubb commented:
@lenzh I have added your suggestion as I did test the following:

- Changed item.created in the database to be a string
- Hit error during the convertQuality method because of it
- Added your patch
- Restarted `ps-inventory` 
- Was able to load the inventory as expected.

RobertGrubb commented:
It's worth noting someone pointed out that the export doesn't work server side with the following:

```
exports['ps-inventory']:OpenInventory(source, name, data)
```

This is actually because if you call the server-side export directly, you also have to pass the source so the export can check the player's state bag:

```
exports['ps-inventory']:OpenInventory(source, name, data, source)
```

lenzh commented:
so you paste the source twice?

RobertGrubb commented:
> so you paste the source twice?

@lenzh Sorry about that, lol, got the params mixed up there.

P1: name ("stash", etc)
P2: id
P3: data
P4: origin (source)

Had to look at it again, the export takes params in a different order than the server's event

lenzh commented:
@RobertGrubb aah oke now i get you

lenzh commented:
also gotta check the commit ur behind with the items going missing in stashes, i never had that issue even with your pr but we might need to bug hunt that

RobertGrubb commented:
@lenzh @xFutte It's good now, thanks for noticing.
